### DISCUSSION TITLE
Video 174: Writing the outline for our first Python Script to setup the data

### DISCUSSION THREAD

**Discussion by sankar82** [mrdbourke/pytorch-deep-learning#616] [Q&A] [answered]
Sec 2.1 I am facing issue with %%writefile (Timestamp 4:30)

I am having trouble using the %%writefile command in Colab.  When I write this is a code cell, the docstring and everything else I type is white in color.  Colab does not recognize any key words.  If I comment out %%writefile, then the cell works correct as normal. I have attached  a screenshot. Please help as I am not able to move forward without fixing this. Thanks and appreciate your help.  

<img width="419" alt="image" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/41609359/46bbf5a2-84b2-4efb-bf88-0edf7ac1d2e0">

**Comment by otrturn** [comment]
Syntax highlighting in colab is not supported when using magic code. I ran into the same issue. At least it does not work for the free version of colab. In a video I sawe it working, but the teacher used a PRO account for colab.

**Reply by sankar82** [reply]
Ah okay, thanks. That's a bummer. Thanks for replying.

**Reply by mrdbourke** [reply, marked as answer]
Thank you for the update @otrturn , I didn't know that Google Colab Pro helped with syntax highlighting.

One option for @sankar82 would be to copy the Python code into a local editor (e.g. VS Code) from Google Colab which will help with code highlighting, however, this is not ideal because of the extra step.

As a side note, my Google Colab (Pro) is now doing a similar thing, all the code in the cell with `%%writefile test_code.py` is white with no highlighting: 

<img width="1624" alt="Screenshot 2023-09-07 at 10 31 05 am" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/16750345/38af3566-2d1c-4252-b38d-e8dc499f82da">

**Reply by sankar82** [reply]
@mrdbourke Thanks for your reply. It is just another Colab thing I guess. Just like how keyboard shortcut to convert code cell to markdown or the shortcut to view docstring don't work anymore. Nevertheless this is not a dealbreaker and I am able to continue with my lessons. Thanks both.

**Comment by DarekW90** [comment]
Same problem here :)
Easyway to fight with this w/o copy code into a local editor is just to comment with # first line, but then remember to uncomment it before run.
Cheers!
